core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -303,6 +303,9 @@ private TableProperties() {}
   public static final String ENGINE_HIVE_ENABLED = "engine.hive.enabled";
   public static final boolean ENGINE_HIVE_ENABLED_DEFAULT = false;
 
+  public static final String LOCK_HIVE_ENABLED = "lock.hive.enabled";
```

Comment:
I used the same pattern than with `engine.hive.enabled`, but if we have a consensus I am happy to use whatever config name is decided upon 😄

Reply:
> Thanks for these efforts @pvary . I left some comments. I think it would be easier to review if we just did the refactor first, then the changes are more minimal for supporting the new NoLock mechanism, which I havent wrapped my mind around fully yet.

Thanks for the review!
Separated the lock refactor to a different PR: #6648, and after it is merged, we can go through the `NoLock` change

Reply:
Renamed